dagmc_stats.py

```diff
@@ -300,3 +300,184 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
+    -------
+    angles : (list) the angles in the tri
+    """
+    
+    angles = []
+    side_lengths = []
+    coord_list = []
+    index = 3
+    if vert != None:
+        verts = my_core.get_adjacencies(tri, 0, op_type=1)
+        for i in range(len(verts)):
+            if verts[i] == vert:
+                index = i
+        for vert in verts:
+            coords = my_core.get_coords(vert)
+            coord_list.append(coords)
+        side_lengths.append(np.linalg.norm(coord_list[index]-coord_list[index-1]))
+        side_lengths.append(np.linalg.norm(coord_list[index]-coord_list[index-2]))
+        side_lengths.append(np.linalg.norm(coord_list[index-1]-coord_list[index-2]))
+    else:
+        side_lengths = get_tri_side_length(my_core, tri)
+    for side in range(3):
+        angles.append(np.arccos((side_lengths[side] * side_lengths[side]
+                            + side_lengths[side - 2] * side_lengths[side - 2]
+                            - side_lengths[side - 1] * side_lengths[side - 1])
+                            /(2.0 * side_lengths[side] * side_lengths[side - 2])))
+    return angles
+
+
+def gaussian_curvature(my_core, vert):
+    """
+    Gets gaussian curvature of a vert
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    vert : vertex
+
+    outputs
+    -------
+    gc : the gaussian curvature of the vert
+    """ 
+    
+    tris = my_core.get_adjacencies(vert, 2)
+    alpha_angles = 0
+    
+    for tri in tris:
+        alpha_angles += get_angles(my_core, tri, vert = vert)[0]
+    
+    gc = np.abs(2*np.pi - alpha_angles)
+    
+    return gc
+
+
+def get_local_roughness(my_core, vert):
+    """
+    Gets local roughness of a vert
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    vert : vertex
+
+    outputs
+    -------
+    lr : the laplacian value of the vert
+    """ 
+    
+    beta_angles = []
+    d = []
+    gc_j = []
+    sum_d_gc = 0
+    
+    gc_i = gaussian_curvature(my_core, vert)
+    
+    adj_tris = my_core.get_adjacencies(vert, 2, op_type=0)
+    
+    #pair tris
+    pair_tris = []
+    tri_vert = []
+    common_vert = []
+    for tri in adj_tris:
+        tri_vert.append(my_core.get_adjacencies(tri, 0, op_type=1))
+    for tri1 in tri_vert:
+        for tri2 in tri_vert:
+            count = 0
+            same_vert = []
+            for m in range(3):
+                for n in range(3):
+                    if tri1[m] == tri2[n]:
+                        count += 1
+                        same_vert.append(tri1[m])
+            if count == 2:
+                pair = []
+                pair.append(tri1)
+                pair.append(tri2)
+                exist = False
+                for p in pair_tris:
+                    if (p[0] == tri1 and p[1] == tri2) or (p[1] == tri1 and p[0] == tri2):
+                        exist = True
+                if not exist:
+                    common_vert.append(same_vert)
+                    pair_tris.append(pair)
```

Comment:
Can you add more comments to this section so that it is easier to follow? Such as why each for loop exists and what the intention is, what you are checking with each if-statement, and what the end goal is of this whole process.